client/src/pages/profile/index.tsx

```diff
@@ -311,8 +311,10 @@ export class ProfilePage extends React.Component<Props, State> {
   };
 
   async componentDidMount() {
-    await this.fetchData();
-    await this.authorizeDiscord();
+    setTimeout(async () => {
```

Comment:
Why do you use here setTimeout ?

Reply:
Thanks. I'll add a comment. It's a hack to fix some next.js caveat that router.query params are not available immediately on the client.